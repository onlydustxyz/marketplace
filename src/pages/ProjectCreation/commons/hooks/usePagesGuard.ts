import { useNavigate } from "react-router-dom";
import { useInformationSession, useOrganizationSession, useResetSession } from "./useProjectCreationSession";
import { RoutePaths } from "src/App";
import { useEffect } from "react";

const STEP_ORGANIZATION = "organization";
const STEP_INFORMATION = "information";
const STEP_REPOSITORY = "repository";

type pageSteps = typeof STEP_ORGANIZATION | typeof STEP_INFORMATION | typeof STEP_REPOSITORY;

export const usePagesGuard = (page: pageSteps) => {
  const { storedValue, status } = useOrganizationSession();
  const { removeValue: removeFormData } = useInformationSession();
  const { reset } = useResetSession();
  const navigate = useNavigate();

  useEffect(() => {
    if ((page === STEP_INFORMATION || page === STEP_REPOSITORY || page === STEP_ORGANIZATION) && status === "getted") {
      if (storedValue?.length === 0) {
        reset();
        navigate(RoutePaths.ProjectCreation);
      } else if (
        page === STEP_INFORMATION &&
        storedValue.filter(organization => (organization.repos || []).find(repo => repo.selected)).length == 0
      ) {
        removeFormData();
        navigate(`${RoutePaths.ProjectCreation}/repository`);
      }
    }
  }, [storedValue, status]);
};
