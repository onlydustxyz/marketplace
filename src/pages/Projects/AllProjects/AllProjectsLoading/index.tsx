import Skeleton from "src/components/Skeleton";

export default function AllProjectLoading() {
  return (
    <div className="flex flex-col space-y-4">
      <Skeleton variant="header" />
      <Skeleton variant="card" />
      <Skeleton variant="card" />
      <Skeleton variant="card" />
      <Skeleton variant="card" />
      <Skeleton variant="card" />
    </div>
  );
}
