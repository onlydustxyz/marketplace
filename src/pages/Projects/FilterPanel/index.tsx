import View from "./View";

type Props = {
  isProjectLeader: boolean;
  fromSidePanel?: boolean;
  technologies: string[];
  sponsors: string[];
};

export default function FilterPanel({ isProjectLeader, fromSidePanel, technologies, sponsors }: Props) {
  return (
    <View
      {...{
        availableTechnologies: technologies,
        availableSponsors: sponsors,
        isProjectLeader,
      }}
      fromSidePanel={fromSidePanel}
    />
  );
}
