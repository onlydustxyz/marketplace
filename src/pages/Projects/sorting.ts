export enum Sorting {
  Trending = "RANK",
  ProjectName = "NAME",
  ReposCount = "REPO_COUNT",
  ContributorsCount = "CONTRIBUTOR_COUNT",
}

export const PROJECT_SORTINGS = [Sorting.Trending, Sorting.ProjectName, Sorting.ReposCount, Sorting.ContributorsCount];
