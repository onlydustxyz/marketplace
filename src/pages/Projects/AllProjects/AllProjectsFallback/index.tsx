import Card from "src/components/Card";
import { useIntl } from "src/hooks/useIntl";
import Button from "src/components/Button";
import { IMAGES } from "src/assets/img";

type Props = {
  clearFilters: () => void;
};

export default function AllProjectsFallback({ clearFilters }: Props) {
  const { T } = useIntl();

  return (
    <Card padded={false} className="flex flex-col items-center gap-8 py-20">
      <img src={IMAGES.global.categories} loading="lazy" alt="Categories" />
      <div className="flex flex-col items-center gap-2">
        <span className="font-belwe text-2xl font-normal text-greyscale-50">{T("projects.fallback.title")}</span>
        <span className="font-walsheim text-base font-normal text-spaceBlue-200">
          {T("projects.fallback.subTitle")}
        </span>
      </div>
      <div onClick={clearFilters}>
        <Button>{T("projects.fallback.clearFiltersButton")}</Button>
      </div>
    </Card>
  );
}
