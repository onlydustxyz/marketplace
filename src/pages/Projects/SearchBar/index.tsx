import { cn } from "src/utils/cn";
import { PropsWithChildren, useRef, useState } from "react";
import { viewportConfig } from "src/config";
import { useIntl } from "src/hooks/useIntl";
import CloseLine from "src/icons/CloseLine";
import SearchLine from "src/icons/SearchLine";
import { useMediaQuery } from "usehooks-ts";
import { useHotkeys } from "react-hotkeys-hook";

type Props = {
  search: string;
  setSearch: (search: string) => void;
};

export default function SearchBar({ search, setSearch }: Props) {
  const { T } = useIntl();
  const isXl = useMediaQuery(`(min-width: ${viewportConfig.breakpoints.xl}px)`);

  const [inputFocus, setInputFocus] = useState(false);

  const textInput = useRef<HTMLInputElement>(null);

  useHotkeys("mod+k", () => {
    textInput.current && textInput.current.focus();
  });
  useHotkeys(
    "esc",
    () => {
      textInput.current && textInput.current.blur();
    },
    { enableOnFormTags: true }
  );

  return (
    <div
      className={cn("overflow-hidden rounded-full p-0.5", {
        "bg-spacePurple-500": inputFocus,
      })}
    >
      <div
        className={cn("relative z-10 flex items-center justify-center rounded-full", {
          "before:absolute before:-z-10 before:h-screen before:w-screen before:scale-x-[8] before:bg-multi-color-gradient before:md:scale-x-[30]":
            !inputFocus,
          "overflow-hidden": !isXl,
        })}
      >
        <ResponsiveOutlineWrapper>
          <div
            className={cn("flex h-12 w-full flex-row items-center gap-2 rounded-full px-4", {
              "bg-spaceBlue-900": !inputFocus,
              "bg-spacePurple-900": inputFocus,
            })}
          >
            <SearchLine
              className={cn("text-2xl", { "text-spaceBlue-200": !inputFocus }, { "text-greyscale-50": inputFocus })}
            />
            <input
              placeholder={T("searchBar.placeholderShort")}
              className="h-8 w-full bg-transparent font-walsheim text-lg text-greyscale-50 outline-none placeholder:text-spaceBlue-200"
              value={search}
              ref={textInput}
              onChange={e => setSearch(e.target.value)}
              onFocus={() => setInputFocus(true)}
              onBlur={() => setInputFocus(false)}
            />
            <button data-testid="clear-searchbar-button">
              <CloseLine
                className={cn(
                  "text-2xl",
                  { hidden: !inputFocus && !search },
                  { "text-greyscale-50": !inputFocus && search },
                  { "text-purple-200": inputFocus }
                )}
                onClick={e => {
                  e.preventDefault();
                  setSearch("");
                }}
              />
            </button>
          </div>
        </ResponsiveOutlineWrapper>
      </div>
    </div>
  );
}

function ResponsiveOutlineWrapper({ children }: PropsWithChildren) {
  const isXl = useMediaQuery(`(min-width: ${viewportConfig.breakpoints.xl}px)`);
  return isXl ? <>{children}</> : <div className="w-full rounded-full border-2 border-transparent">{children}</div>;
}
