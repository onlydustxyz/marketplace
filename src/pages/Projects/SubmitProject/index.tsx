import { useNavigate } from "react-router-dom";
import { RoutePaths } from "src/App";
import Button, { ButtonSize, ButtonType, Width } from "src/components/Button";
import Card from "src/components/Card";
import { useAuth } from "src/hooks/useAuth";
import { useIntl } from "src/hooks/useIntl";
import { SessionMethod, useSessionDispatch } from "src/hooks/useSession";
import config from "src/config";

export const LOGIN_URL = `${config.LOGIN_URL}?redirect_url=${encodeURI(window.location.origin)}`;

export default function SubmitProject() {
  const { T } = useIntl();
  const { isLoggedIn } = useAuth();
  const navigate = useNavigate();

  const dispatchSession = useSessionDispatch();

  const startProjectCreation = () => {
    if (isLoggedIn) {
      navigate(RoutePaths.ProjectCreation);
    } else {
      dispatchSession({ method: SessionMethod.SetVisitedPageBeforeLogin, value: RoutePaths.ProjectCreation });
      window.location.replace(LOGIN_URL);
    }
  };

  return (
    <Card className="mb-4 flex h-fit flex-row gap-4 p-6">
      <div className="flex-1 text-sm leading-4">{T("project.details.create.description")}</div>
      <Button
        htmlType="submit"
        size={ButtonSize.Sm}
        type={ButtonType.Primary}
        width={Width.Fit}
        onClick={() => startProjectCreation()}
      >
        <i className="ri-magic-line" />
        {T("project.details.create.submit.button")}
      </Button>
    </Card>
  );
}
