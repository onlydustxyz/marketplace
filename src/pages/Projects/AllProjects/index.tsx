import { useMemo, useEffect } from "react";
import ProjectCard from "src/components/ProjectCard";
import { useProjectFilter } from "src/pages/Projects/useProjectFilter";
import AllProjectsFallback from "./AllProjectsFallback";
import SortingDropdown, { PROJECT_SORTINGS, Sorting } from "src/pages/Projects/Sorting/SortingDropdown";
import { useIntl } from "src/hooks/useIntl";
import { FilterButton } from "src/pages/Projects/FilterPanel/FilterButton";
import { SortButton } from "src/pages/Projects/Sorting/SortButton";
import { ApiResourcePaths } from "src/App/DataWrapper/config";
import { useRestfulData } from "src/hooks/useRestfulData";
import ErrorFallback from "src/ErrorFallback";
import AllProjectLoading from "./AllProjectsLoading";
import { Project } from "src/types";

export const DEFAULT_SORTING = Sorting.Trending;

type Props = {
  search: string;
  clearSearch: () => void;
  sorting?: Sorting;
  setSorting: (sorting?: Sorting) => void;
  restoreScroll: () => void;
  filterPanelOpen: boolean;
  setFilterPanelOpen: (open: boolean) => void;
  sortingPanelOpen: boolean;
  setSortingPanelOpen: (open: boolean) => void;
  setTechnologies: (technologies: string[]) => void;
  setSponsors: (sponsors: string[]) => void;
};

export default function AllProjects({
  search,
  clearSearch,
  sorting,
  setSorting,
  restoreScroll,
  filterPanelOpen,
  setFilterPanelOpen,
  sortingPanelOpen,
  setSortingPanelOpen,
  setTechnologies,
  setSponsors,
}: Props) {
  const { T } = useIntl();

  const {
    projectFilter: { ownership, technologies, sponsors },
    clear: clearFilters,
  } = useProjectFilter();

  const queryParams = useMemo(
    () => [
      ...(technologies.length > 0 ? [{ key: "technologies", value: technologies }] : []),
      ...(sponsors.length > 0 ? [{ key: "sponsor", value: sponsors }] : []),
      ...(search ? [{ key: "search", value: [search] }] : []),
      ...(sorting ? [{ key: "sort", value: [sorting] }] : []),
      ...(ownership ? [{ key: "mine", value: [ownership === "Mine" ? true : false] }] : []),
    ],
    [technologies, sponsors, search, sorting, ownership]
  );

  const { data, isLoading, isError } = useRestfulData({
    resourcePath: ApiResourcePaths.GET_ALL_PROJECTS,
    queryParams,
    method: "GET",
  });

  useEffect(() => {
    restoreScroll();
  }, [restoreScroll]);

  useEffect(() => {
    if (data && !isLoading) {
      const { technologies, sponsors } = data;
      setTechnologies(replaceApostrophes(technologies) || []);
      setSponsors(sponsors || []);
    }
  }, [data]);

  if (isLoading) {
    return <AllProjectLoading />;
  }

  // if there is an error we need to return ErrorFallback component
  if (isError) {
    return <ErrorFallback />;
  }

  const { projects } = data || { projects: [] };

  // if projects is empty and loading is false, we need to return a fallback
  if (projects.length === 0 && !isError && !isLoading) {
    return (
      <AllProjectsFallback
        clearFilters={() => {
          clearFilters();
          clearSearch();
        }}
      />
    );
  }

  return (
    <div className="flex flex-col gap-5">
      <div className="relative flex h-10 items-center justify-between">
        <div className="px-2 font-medium text-spaceBlue-200">{T("projects.count", { count: projects.length })}</div>
        <div className="absolute right-0 top-0 z-10 hidden xl:block">
          <SortingDropdown all={PROJECT_SORTINGS} current={sorting || DEFAULT_SORTING} onChange={setSorting} />
        </div>
        <div className="flex items-center gap-2 xl:hidden">
          <SortButton panelOpen={sortingPanelOpen} setPanelOpen={setSortingPanelOpen} />
          <FilterButton panelOpen={filterPanelOpen} setPanelOpen={setFilterPanelOpen} />
        </div>
      </div>
      <div className="flex grow flex-col gap-5">
        {projects.map((project: Project, index: number) => {
          const isFirstHiringProject = index === 0 && project.hiring;
          return (
            <ProjectCard className={isFirstHiringProject ? "mt-3" : undefined} key={project.id} project={project} />
          );
        })}
      </div>
    </div>
  );
}

function replaceApostrophes(array: string[]): string[] {
  return array.map(item => item.replace(/'/g, " "));
}
