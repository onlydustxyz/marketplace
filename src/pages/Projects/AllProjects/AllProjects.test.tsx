import React from "react";
import { render } from "@testing-library/react";
import AllProjects from "."; // Adjust the import to your file structure
import { BrowserRouter as Router } from "react-router-dom";
import { QueryClient, QueryClientProvider } from "react-query";
import { TokenSetProvider } from "src/hooks/useTokenSet";
import { ProjectFilterProvider } from "src/pages/Projects/useProjectFilter";

// Create a client
const queryClient = new QueryClient();

jest.mock("src/pages/Projects/useProjectFilter", () => ({
  useProjectFilter: () => ({
    projectFilter: { ownership: "", technologies: [], sponsors: [] },
    clear: jest.fn(),
  }),
}));

jest.mock("src/hooks/useRestfulData", () => ({
  useRestfulData: () => ({
    data: {
      projects: [],
      technologies: [],
      sponsors: [],
    },
    isLoading: false,
    isError: false,
  }),
}));

describe("<AllProjects />", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <Router>
        <TokenSetProvider>
          <ProjectFilterProvider>
            <QueryClientProvider client={queryClient}>
              <AllProjects
                search=""
                clearSearch={jest.fn()}
                sorting={undefined}
                setSorting={jest.fn()}
                restoreScroll={jest.fn()}
                filterPanelOpen={false}
                setFilterPanelOpen={jest.fn()}
                sortingPanelOpen={false}
                setSortingPanelOpen={jest.fn()}
                setTechnologies={jest.fn()}
                setSponsors={jest.fn()}
              />
            </QueryClientProvider>
          </ProjectFilterProvider>
        </TokenSetProvider>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
