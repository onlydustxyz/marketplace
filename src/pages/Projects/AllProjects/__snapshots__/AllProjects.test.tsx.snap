// Vitest Snapshot v1

exports[`<AllProjects /> > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col space-y-4"
  >
    <div
      class="relative w-full"
    >
      <svg
        class="relative w-full animate-pulse"
        fill="none"
        viewBox="0 0 1440 76"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_2_76)"
        >
          <path
            d="M1440 0H0V76H1440V0Z"
            fill="black"
          />
          <path
            d="M44 58C55.0457 58 64 49.0457 64 38C64 26.9543 55.0457 18 44 18C32.9543 18 24 26.9543 24 38C24 49.0457 32.9543 58 44 58Z"
            fill="#171D44"
          />
          <path
            d="M174 26H88C81.3726 26 76 31.3726 76 38C76 44.6274 81.3726 50 88 50H174C180.627 50 186 44.6274 186 38C186 31.3726 180.627 26 174 26Z"
            fill="#171D44"
          />
          <path
            d="M1394 16H1311C1298.85 16 1289 25.8497 1289 38C1289 50.1503 1298.85 60 1311 60H1394C1406.15 60 1416 50.1503 1416 38C1416 25.8497 1406.15 16 1394 16Z"
            fill="#171D44"
          />
        </g>
        <defs>
          <clippath
            id="clip0_2_76"
          >
            <rect
              fill="white"
              height="76"
              width="1440"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <div
      class="relative w-full"
    >
      <svg
        class="relative w-full animate-pulse"
        fill="none"
        viewBox="0 0 816 144"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_2_24)"
        >
          <mask
            height="144"
            id="mask0_2_24"
            maskUnits="userSpaceOnUse"
            width="816"
            x="0"
            y="0"
          >
            <path
              d="M800 0H16C7.16344 0 0 7.16344 0 16V128C0 136.837 7.16344 144 16 144H800C808.837 144 816 136.837 816 128V16C816 7.16344 808.837 0 800 0Z"
              fill="white"
            />
          </mask>
          <g
            mask="url(#mask0_2_24)"
          >
            <path
              d="M800 0H16C7.16344 0 0 7.16344 0 16V128C0 136.837 7.16344 144 16 144H800C808.837 144 816 136.837 816 128V16C816 7.16344 808.837 0 800 0Z"
              fill="#0E0D2E"
            />
            <path
              d="M60 22H36C28.268 22 22 28.268 22 36V60C22 67.732 28.268 74 36 74H60C67.732 74 74 67.732 74 60V36C74 28.268 67.732 22 60 22Z"
              fill="#171D44"
            />
            <path
              d="M60 22H36C28.268 22 22 28.268 22 36V60C22 67.732 28.268 74 36 74H60C67.732 74 74 67.732 74 60V36C74 28.268 67.732 22 60 22Z"
              stroke="#171D44"
              stroke-width="4"
            />
            <path
              d="M276 24H100C93.3726 24 88 29.3726 88 36C88 42.6274 93.3726 48 100 48H276C282.627 48 288 42.6274 288 36C288 29.3726 282.627 24 276 24Z"
              fill="#171D44"
            />
            <path
              d="M232 56H96C91.5817 56 88 59.5817 88 64C88 68.4183 91.5817 72 96 72H232C236.418 72 240 68.4183 240 64C240 59.5817 236.418 56 232 56Z"
              fill="#171D44"
            />
            <path
              d="M147 88H40C31.1634 88 24 95.1634 24 104C24 112.837 31.1634 120 40 120H147C155.837 120 163 112.837 163 104C163 95.1634 155.837 88 147 88Z"
              fill="#171D44"
            />
            <path
              d="M312 24V120"
              stroke="#171D44"
            />
            <path
              d="M784 28H344C339.582 28 336 31.5817 336 36C336 40.4183 339.582 44 344 44H784C788.418 44 792 40.4183 792 36C792 31.5817 788.418 28 784 28Z"
              fill="#171D44"
            />
            <path
              d="M545 52H344C339.582 52 336 55.5817 336 60C336 64.4183 339.582 68 344 68H545C549.418 68 553 64.4183 553 60C553 55.5817 549.418 52 545 52Z"
              fill="#171D44"
            />
            <path
              d="M427 92H348C341.373 92 336 97.3726 336 104C336 110.627 341.373 116 348 116H427C433.627 116 439 110.627 439 104C439 97.3726 433.627 92 427 92Z"
              fill="#171D44"
            />
            <path
              d="M538 92H459C452.373 92 447 97.3726 447 104C447 110.627 452.373 116 459 116H538C544.627 116 550 110.627 550 104C550 97.3726 544.627 92 538 92Z"
              fill="#171D44"
            />
            <path
              d="M649 92H570C563.373 92 558 97.3726 558 104C558 110.627 563.373 116 570 116H649C655.627 116 661 110.627 661 104C661 97.3726 655.627 92 649 92Z"
              fill="#171D44"
            />
          </g>
        </g>
        <defs>
          <clippath
            id="clip0_2_24"
          >
            <rect
              fill="white"
              height="144"
              width="816"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <div
      class="relative w-full"
    >
      <svg
        class="relative w-full animate-pulse"
        fill="none"
        viewBox="0 0 816 144"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_2_24)"
        >
          <mask
            height="144"
            id="mask0_2_24"
            maskUnits="userSpaceOnUse"
            width="816"
            x="0"
            y="0"
          >
            <path
              d="M800 0H16C7.16344 0 0 7.16344 0 16V128C0 136.837 7.16344 144 16 144H800C808.837 144 816 136.837 816 128V16C816 7.16344 808.837 0 800 0Z"
              fill="white"
            />
          </mask>
          <g
            mask="url(#mask0_2_24)"
          >
            <path
              d="M800 0H16C7.16344 0 0 7.16344 0 16V128C0 136.837 7.16344 144 16 144H800C808.837 144 816 136.837 816 128V16C816 7.16344 808.837 0 800 0Z"
              fill="#0E0D2E"
            />
            <path
              d="M60 22H36C28.268 22 22 28.268 22 36V60C22 67.732 28.268 74 36 74H60C67.732 74 74 67.732 74 60V36C74 28.268 67.732 22 60 22Z"
              fill="#171D44"
            />
            <path
              d="M60 22H36C28.268 22 22 28.268 22 36V60C22 67.732 28.268 74 36 74H60C67.732 74 74 67.732 74 60V36C74 28.268 67.732 22 60 22Z"
              stroke="#171D44"
              stroke-width="4"
            />
            <path
              d="M276 24H100C93.3726 24 88 29.3726 88 36C88 42.6274 93.3726 48 100 48H276C282.627 48 288 42.6274 288 36C288 29.3726 282.627 24 276 24Z"
              fill="#171D44"
            />
            <path
              d="M232 56H96C91.5817 56 88 59.5817 88 64C88 68.4183 91.5817 72 96 72H232C236.418 72 240 68.4183 240 64C240 59.5817 236.418 56 232 56Z"
              fill="#171D44"
            />
            <path
              d="M147 88H40C31.1634 88 24 95.1634 24 104C24 112.837 31.1634 120 40 120H147C155.837 120 163 112.837 163 104C163 95.1634 155.837 88 147 88Z"
              fill="#171D44"
            />
            <path
              d="M312 24V120"
              stroke="#171D44"
            />
            <path
              d="M784 28H344C339.582 28 336 31.5817 336 36C336 40.4183 339.582 44 344 44H784C788.418 44 792 40.4183 792 36C792 31.5817 788.418 28 784 28Z"
              fill="#171D44"
            />
            <path
              d="M545 52H344C339.582 52 336 55.5817 336 60C336 64.4183 339.582 68 344 68H545C549.418 68 553 64.4183 553 60C553 55.5817 549.418 52 545 52Z"
              fill="#171D44"
            />
            <path
              d="M427 92H348C341.373 92 336 97.3726 336 104C336 110.627 341.373 116 348 116H427C433.627 116 439 110.627 439 104C439 97.3726 433.627 92 427 92Z"
              fill="#171D44"
            />
            <path
              d="M538 92H459C452.373 92 447 97.3726 447 104C447 110.627 452.373 116 459 116H538C544.627 116 550 110.627 550 104C550 97.3726 544.627 92 538 92Z"
              fill="#171D44"
            />
            <path
              d="M649 92H570C563.373 92 558 97.3726 558 104C558 110.627 563.373 116 570 116H649C655.627 116 661 110.627 661 104C661 97.3726 655.627 92 649 92Z"
              fill="#171D44"
            />
          </g>
        </g>
        <defs>
          <clippath
            id="clip0_2_24"
          >
            <rect
              fill="white"
              height="144"
              width="816"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <div
      class="relative w-full"
    >
      <svg
        class="relative w-full animate-pulse"
        fill="none"
        viewBox="0 0 816 144"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_2_24)"
        >
          <mask
            height="144"
            id="mask0_2_24"
            maskUnits="userSpaceOnUse"
            width="816"
            x="0"
            y="0"
          >
            <path
              d="M800 0H16C7.16344 0 0 7.16344 0 16V128C0 136.837 7.16344 144 16 144H800C808.837 144 816 136.837 816 128V16C816 7.16344 808.837 0 800 0Z"
              fill="white"
            />
          </mask>
          <g
            mask="url(#mask0_2_24)"
          >
            <path
              d="M800 0H16C7.16344 0 0 7.16344 0 16V128C0 136.837 7.16344 144 16 144H800C808.837 144 816 136.837 816 128V16C816 7.16344 808.837 0 800 0Z"
              fill="#0E0D2E"
            />
            <path
              d="M60 22H36C28.268 22 22 28.268 22 36V60C22 67.732 28.268 74 36 74H60C67.732 74 74 67.732 74 60V36C74 28.268 67.732 22 60 22Z"
              fill="#171D44"
            />
            <path
              d="M60 22H36C28.268 22 22 28.268 22 36V60C22 67.732 28.268 74 36 74H60C67.732 74 74 67.732 74 60V36C74 28.268 67.732 22 60 22Z"
              stroke="#171D44"
              stroke-width="4"
            />
            <path
              d="M276 24H100C93.3726 24 88 29.3726 88 36C88 42.6274 93.3726 48 100 48H276C282.627 48 288 42.6274 288 36C288 29.3726 282.627 24 276 24Z"
              fill="#171D44"
            />
            <path
              d="M232 56H96C91.5817 56 88 59.5817 88 64C88 68.4183 91.5817 72 96 72H232C236.418 72 240 68.4183 240 64C240 59.5817 236.418 56 232 56Z"
              fill="#171D44"
            />
            <path
              d="M147 88H40C31.1634 88 24 95.1634 24 104C24 112.837 31.1634 120 40 120H147C155.837 120 163 112.837 163 104C163 95.1634 155.837 88 147 88Z"
              fill="#171D44"
            />
            <path
              d="M312 24V120"
              stroke="#171D44"
            />
            <path
              d="M784 28H344C339.582 28 336 31.5817 336 36C336 40.4183 339.582 44 344 44H784C788.418 44 792 40.4183 792 36C792 31.5817 788.418 28 784 28Z"
              fill="#171D44"
            />
            <path
              d="M545 52H344C339.582 52 336 55.5817 336 60C336 64.4183 339.582 68 344 68H545C549.418 68 553 64.4183 553 60C553 55.5817 549.418 52 545 52Z"
              fill="#171D44"
            />
            <path
              d="M427 92H348C341.373 92 336 97.3726 336 104C336 110.627 341.373 116 348 116H427C433.627 116 439 110.627 439 104C439 97.3726 433.627 92 427 92Z"
              fill="#171D44"
            />
            <path
              d="M538 92H459C452.373 92 447 97.3726 447 104C447 110.627 452.373 116 459 116H538C544.627 116 550 110.627 550 104C550 97.3726 544.627 92 538 92Z"
              fill="#171D44"
            />
            <path
              d="M649 92H570C563.373 92 558 97.3726 558 104C558 110.627 563.373 116 570 116H649C655.627 116 661 110.627 661 104C661 97.3726 655.627 92 649 92Z"
              fill="#171D44"
            />
          </g>
        </g>
        <defs>
          <clippath
            id="clip0_2_24"
          >
            <rect
              fill="white"
              height="144"
              width="816"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <div
      class="relative w-full"
    >
      <svg
        class="relative w-full animate-pulse"
        fill="none"
        viewBox="0 0 816 144"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_2_24)"
        >
          <mask
            height="144"
            id="mask0_2_24"
            maskUnits="userSpaceOnUse"
            width="816"
            x="0"
            y="0"
          >
            <path
              d="M800 0H16C7.16344 0 0 7.16344 0 16V128C0 136.837 7.16344 144 16 144H800C808.837 144 816 136.837 816 128V16C816 7.16344 808.837 0 800 0Z"
              fill="white"
            />
          </mask>
          <g
            mask="url(#mask0_2_24)"
          >
            <path
              d="M800 0H16C7.16344 0 0 7.16344 0 16V128C0 136.837 7.16344 144 16 144H800C808.837 144 816 136.837 816 128V16C816 7.16344 808.837 0 800 0Z"
              fill="#0E0D2E"
            />
            <path
              d="M60 22H36C28.268 22 22 28.268 22 36V60C22 67.732 28.268 74 36 74H60C67.732 74 74 67.732 74 60V36C74 28.268 67.732 22 60 22Z"
              fill="#171D44"
            />
            <path
              d="M60 22H36C28.268 22 22 28.268 22 36V60C22 67.732 28.268 74 36 74H60C67.732 74 74 67.732 74 60V36C74 28.268 67.732 22 60 22Z"
              stroke="#171D44"
              stroke-width="4"
            />
            <path
              d="M276 24H100C93.3726 24 88 29.3726 88 36C88 42.6274 93.3726 48 100 48H276C282.627 48 288 42.6274 288 36C288 29.3726 282.627 24 276 24Z"
              fill="#171D44"
            />
            <path
              d="M232 56H96C91.5817 56 88 59.5817 88 64C88 68.4183 91.5817 72 96 72H232C236.418 72 240 68.4183 240 64C240 59.5817 236.418 56 232 56Z"
              fill="#171D44"
            />
            <path
              d="M147 88H40C31.1634 88 24 95.1634 24 104C24 112.837 31.1634 120 40 120H147C155.837 120 163 112.837 163 104C163 95.1634 155.837 88 147 88Z"
              fill="#171D44"
            />
            <path
              d="M312 24V120"
              stroke="#171D44"
            />
            <path
              d="M784 28H344C339.582 28 336 31.5817 336 36C336 40.4183 339.582 44 344 44H784C788.418 44 792 40.4183 792 36C792 31.5817 788.418 28 784 28Z"
              fill="#171D44"
            />
            <path
              d="M545 52H344C339.582 52 336 55.5817 336 60C336 64.4183 339.582 68 344 68H545C549.418 68 553 64.4183 553 60C553 55.5817 549.418 52 545 52Z"
              fill="#171D44"
            />
            <path
              d="M427 92H348C341.373 92 336 97.3726 336 104C336 110.627 341.373 116 348 116H427C433.627 116 439 110.627 439 104C439 97.3726 433.627 92 427 92Z"
              fill="#171D44"
            />
            <path
              d="M538 92H459C452.373 92 447 97.3726 447 104C447 110.627 452.373 116 459 116H538C544.627 116 550 110.627 550 104C550 97.3726 544.627 92 538 92Z"
              fill="#171D44"
            />
            <path
              d="M649 92H570C563.373 92 558 97.3726 558 104C558 110.627 563.373 116 570 116H649C655.627 116 661 110.627 661 104C661 97.3726 655.627 92 649 92Z"
              fill="#171D44"
            />
          </g>
        </g>
        <defs>
          <clippath
            id="clip0_2_24"
          >
            <rect
              fill="white"
              height="144"
              width="816"
            />
          </clippath>
        </defs>
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
